classes/plugins/ThemePlugin.inc.php

```diff
@@ -59,18 +76,40 @@ function register($category, $path) {
 			return true;
 		}
 
-		// Fire an initialization method which themes should use to add
-		// styles, scripts and fonts
-		$this->init();
+		// Themes must initialize their functionality after all theme plugins
+		// have been loaded in order to make use of parent/child theme
+		// relationships
+		HookRegistry::register('PluginRegistry::categoryLoaded::themes', array($this, 'themeRegistered'));
+		HookRegistry::register('PluginRegistry::categoryLoaded::themes', array($this, 'initAfter'));
 
-		$this->_registerTemplates();
-		$this->_registerStyles();
-		$this->_registerScripts();
+		// Save any theme options displayed on the appearance and site settings
+		// forms
+		HookRegistry::register('appearanceform::execute', array($this, 'saveOptionsForm'));
+		HookRegistry::register('appearanceform::readuservars', array($this, 'readOptionsFormUserVars'));
+		HookRegistry::register('sitesetupform::execute', array($this, 'saveOptionsForm'));
+		HookRegistry::register('sitesetupform::readuservars', array($this, 'readOptionsFormUserVars'));
 
 		return true;
 	}
 
 	/**
+	 * Fire the init() method when a theme is registered
+	 *
+	 * @param $themes array List of all loaded themes
+	 * @return null
+	 */
+	public function themeRegistered($themes) {
```

Comment:
I'm not super fond of the names `themeRegistered` or `initAfter` -- but my imagination is slow or absent, so I'm content to leave them for now.